src/main/java/net/fabricmc/loom/LoomGradlePlugin.java

```diff
@@ -24,118 +24,63 @@
 
 package net.fabricmc.loom;
 
-import java.io.File;
-import java.util.Locale;
-
+import com.google.common.collect.ImmutableMap;
+import org.gradle.api.Plugin;
 import org.gradle.api.Project;
-import org.gradle.api.tasks.TaskContainer;
+import org.gradle.plugins.ide.idea.model.IdeaModel;
 
-import net.fabricmc.loom.api.decompilers.LoomDecompiler;
+import net.fabricmc.loom.configuration.CompileConfiguration;
+import net.fabricmc.loom.configuration.FabricApiExtension;
+import net.fabricmc.loom.configuration.MavenPublication;
+import net.fabricmc.loom.configuration.providers.mappings.MappingsCache;
 import net.fabricmc.loom.decompilers.cfr.FabricCFRDecompiler;
 import net.fabricmc.loom.decompilers.fernflower.FabricFernFlowerDecompiler;
-import net.fabricmc.loom.providers.MappingsProvider;
-import net.fabricmc.loom.task.CleanEclipseRunsTask;
-import net.fabricmc.loom.task.CleanLoomBinaries;
-import net.fabricmc.loom.task.CleanLoomMappings;
-import net.fabricmc.loom.task.DownloadAssetsTask;
-import net.fabricmc.loom.task.GenEclipseRunsTask;
-import net.fabricmc.loom.task.GenIdeaProjectTask;
-import net.fabricmc.loom.task.GenVsCodeProjectTask;
-import net.fabricmc.loom.task.GenerateSourcesTask;
-import net.fabricmc.loom.task.MigrateMappingsTask;
-import net.fabricmc.loom.task.RemapJarTask;
-import net.fabricmc.loom.task.RemapSourcesJarTask;
-import net.fabricmc.loom.task.RunClientTask;
-import net.fabricmc.loom.task.RunServerTask;
-
-public class LoomGradlePlugin extends AbstractPlugin {
-	public static File getMappedByproduct(Project project, String suffix) {
-		LoomGradleExtension extension = project.getExtensions().getByType(LoomGradleExtension.class);
-		MappingsProvider mappingsProvider = extension.getMappingsProvider();
-		File mappedJar = mappingsProvider.mappedProvider.getMappedJar();
-		String path = mappedJar.getAbsolutePath();
-
-		if (!path.toLowerCase(Locale.ROOT).endsWith(".jar")) {
-			throw new RuntimeException("Invalid mapped JAR path: " + path);
-		}
-
-		return new File(path.substring(0, path.length() - 4) + suffix);
-	}
+import net.fabricmc.loom.task.LoomTasks;
+import net.fabricmc.loom.util.DownloadUtil;
 
+public class LoomGradlePlugin implements Plugin<Project> {
 	@Override
-	public void apply(Project target) {
-		super.apply(target);
-
-		TaskContainer tasks = target.getTasks();
+	public void apply(Project project) {
+		project.getLogger().lifecycle("Fabric Loom: " + LoomGradlePlugin.class.getPackage().getImplementationVersion());
 
-		tasks.register("cleanLoomBinaries", CleanLoomBinaries.class, t -> t.setDescription("Removes binary jars created by Loom."));
-		tasks.register("cleanLoomMappings", CleanLoomMappings.class, t -> t.setDescription("Removes mappings downloaded by Loom."));
+		boolean refreshDeps = project.getGradle().getStartParameter().isRefreshDependencies();
+		DownloadUtil.refreshDeps = refreshDeps;
```

Comment:
Might be cleaner to just refer to a getter in loom's plugin or the extension instead of a public field on DownloadUtil?

Reply:
Yes, ideally we would pass the project into download util, but that will do for now.